src/ansys/mechanical/core/embedding/app.py

```diff
@@ -194,9 +196,46 @@ def save(self, path=None):
         else:
             self.DataModel.Project.Save()
 
-    def save_as(self, path):
-        """Save the project as."""
-        self.DataModel.Project.SaveAs(path)
+    def save_as(self, path: str, overwrite: bool = False):
+        """Save the project as a new file.
+
+        If the `overwrite` flag is enabled, the current saved file is temporarily moved
+        to a backup location. The new file is then saved in its place. If the process fails,
+        the backup file is restored to its original location.
+
+        Parameters
+        ----------
+        path: int, optional
+            The path where file needs to be saved.
+        overwrite: bool, optional
+            Whether the file should be overwritten if it already exists (default is False).
+        """
+        if not os.path.exists(path):
+            self.DataModel.Project.SaveAs(path)
+        else:
+            if not overwrite:
+                raise Exception(
+                    f"File already exists in {path}, Use ``overwrite`` flag to "
+                    "replace the existing file."
+                )
+
+            temp_dir = tempfile.mkdtemp()
+            temp_file_path = os.path.join(temp_dir, os.path.basename(path))
+
+            try:
+                # Move existing file to temp location
+                shutil.move(path, temp_file_path)
```

Comment:
@dipinknair why don't we just delete the files in path and save_as to it?

Reply:
In case if it fails to do a save_as, we should not be losing the existing saved files right ?

Reply:
I'm fine with it. The user said to overwrite, I suppose they must not care too much about losing the files they ask to overwrite

Reply:
I updated code so that it deletes existing files before saving as new.